src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -98,6 +102,32 @@ public TripPatternForDate(
           .asDateTime(localDate, last.getArrivalTime(last.getNumStops() - 1))
           .toLocalDate();
     }
+    if (startOfRunningPeriod.isAfter(endOfRunningPeriod)) {
+      if (hasFrequencies()) {
+        var startTripId = frequencies.get(0).tripTimes.getTrip().getId();
+        var endTripId = frequencies.get(frequencies.size() - 1).tripTimes.getTrip().getId();
+        LOG.warn(
+          "Could not construct as start of the running period {} in frequency trip {} is after the end {} in frequency trip {}",
+          startOfRunningPeriod,
+          startTripId,
+          endOfRunningPeriod,
+          endTripId
+        );
+      } else {
+        var startTripId = tripTimes.get(0).getTrip().getId();
+        var endTripId = tripTimes.get(tripTimes.size() - 1).getTrip().getId();
+        LOG.warn(
+          "Could not construct as start of the running period {} in trip {} is after the end {} in trip {}",
+          startOfRunningPeriod,
+          startTripId,
+          endOfRunningPeriod,
+          endTripId
+        );
+      }
+      throw new IllegalArgumentException(
+        "Start of the running period is after end of the running period"
+      );
+    }
```

Comment:
This is not DRY, add a private static assertXyz method and call it from within the two blocks above.